packages/amplify-category-auth/provider-utils/awscloudformation/triggers/PostConfirmation/add-to-group.js

```diff
@@ -1,34 +1,30 @@
 /* eslint-disable-line */ const aws = require('aws-sdk');
 
-exports.handler = async (event, context) => {
-  const cognitoidentityserviceprovider = new aws.CognitoIdentityServiceProvider({ apiVersion: '2016-04-18' });
+exports.handler = async event => {
+  const cognitoidentityserviceprovider = new aws.CognitoIdentityServiceProvider({
+    apiVersion: '2016-04-18',
+  });
   const groupParams = {
     GroupName: process.env.GROUP,
     UserPoolId: event.userPoolId,
   };
-
   const addUserParams = {
     GroupName: process.env.GROUP,
     UserPoolId: event.userPoolId,
     Username: event.userName,
   };
-
+  /**
+   * Check if the group exists; if it doesn't, create it.
+   */
   try {
     await cognitoidentityserviceprovider.getGroup(groupParams).promise();
   } catch (e) {
     await cognitoidentityserviceprovider.createGroup(groupParams).promise();
   }
+  /**
+   * Then, add the user to the group.
+   */
+  await cognitoidentityserviceprovider.adminAddUserToGroup(addUserParams).promise();
 
-  try {
-    await cognitoidentityserviceprovider.adminAddUserToGroup(addUserParams).promise();
-    return {
-      statusCode: 200,
-      body: JSON.stringify(event)
-    }
-  } catch (error) {
-    return {
-      statusCode: 500,
-      body: JSON.stringify(error)
-    }
-  }
+  return event;
```

Comment:
Just saw this - I don't think so. These changes were needed to get the service working for me. It was completely broken before, it would return before waiting for the result, so it can't be *more* broken in any external parts than it was before, so that could be followed up on independently IMO.

I linked the [ctjlewis/amplify-dev](https://github.com/ctjlewis/amplify-dev) repo for testing.